roles/core/hosts_file/templates/hosts.j2

```diff
@@ -9,71 +9,71 @@
+        {% if (global_network_settings is defined and global_network_settings is not none) and (global_network_settings.enable_global_network is defined and global_network_settings.enable_global_network is not none) and (global_network_settings.global_network_group is defined and global_network_settings.global_network_group is not none) and (host in groups[global_network_settings.global_network_group]) %}
           {% if interface_vars.network == global_network_settings.global_network %}
-{{interface_vars.ip4}} {{host}} {{host}}.{{domain_name}} {{host}}-{{interface_vars.network}} {{alias_list|join(' ')}}
+{{ interface_vars.ip4 }} {{ host }} {{ host }}.{{ domain_name }} {{ host }}-{{ interface_vars.network }} {{ alias_list|join(' ') }}
           {% else %}
-{{interface_vars.ip4}} {{host}}-{{interface_vars.network}}
+{{ interface_vars.ip4 }} {{ host }}-{{ interface_vars.network }}
           {% endif %}
         {% else %}
           {% if interface == host_node_main_network_interface %}
-{{interface_vars.ip4}} {{host}} {{host}}.{{domain_name}} {{host}}-{{interface_vars.network}} {{alias_list|join(' ')}}
+{{ interface_vars.ip4 }} {{ host }} {{ host }}.{{ domain_name }} {{ host }}-{{ interface_vars.network }} {{ alias_list|join(' ') }}
           {% else %}
-{{interface_vars.ip4}} {{host}}-{{interface_vars.network}}
+{{ interface_vars.ip4 }} {{ host }}-{{ interface_vars.network }}
           {% endif %}
         {% endif %}
       {% endif %}
     {% endfor %}
   {% endif %}
 
   {#- Write BMC #}
-  {% if host_dict['bmc'] is defined and not none %}
+  {% if host_dict['bmc'] is defined and host_dict['bmc'] is not none %}
 
   {#- Gather BMC global alias and alias (icebergs related) #}
     {% set bmc_alias_list = [] %}
-    {% if host_dict['bmc']['global_alias'] is defined and not none %}
+    {% if host_dict['bmc']['global_alias'] is defined and host_dict['bmc']['global_alias'] is not none %}
       {% for alias in host_dict['bmc']['global_alias'] %}{{ bmc_alias_list.append(alias) }}{% endfor %}
     {% endif %}
-    {% if host_dict['bmc']['alias'] is defined and not none and host_iceberg == j2_current_iceberg %}
+    {% if host_dict['bmc']['alias'] is defined and host_dict['bmc']['alias'] is not none and host_iceberg == j2_current_iceberg %}
       {% for alias in host_dict['bmc']['alias'] %}{{ bmc_alias_list.append(alias) }}{% endfor %}
     {% endif %}
 
-    {%- if (host_dict['bmc'].name is defined and not none) and (host_dict['bmc'].ip4 is defined and not none) %}
- {{host_dict['bmc'].ip4}} {{host_dict['bmc'].name}} {{bmc_alias_list|join(' ')}}
+    {%- if (host_dict['bmc'].name is defined and host_dict['bmc'].name is not none) and (host_dict['bmc'].ip4 is defined and host_dict['bmc'].ip4 is not none) %}
+ {{ host_dict['bmc'].ip4 }} {{ host_dict['bmc'].name }} {{ bmc_alias_list|join(' ') }}
```

Comment:
Can you remove the leading whitespace for bmc?

```suggestion
{{ host_dict['bmc'].ip4 }} {{ host_dict['bmc'].name }} {{ bmc_alias_list|join(' ') }}
```

It will produce:
```
TASK [hosts_file : Template >> /etc/hosts] ***************************
--- before: /etc/hosts
+++ after: /root/.ansible/tmp/ansible-local-290445fgv3af7/tmpor5ujxlc/hosts.j2
@@ -11,7 +11,7 @@

 10.10.0.1 management1 management1.tumulus.tld management1-ice1-1
 10.20.0.1 management1-interconnect-1
- 10.10.100.1 bmanagement1 
+10.10.100.1 bmanagement1 

 10.10.3.1 c001 c001.tumulus.tld c001-ice1-1
 10.20.3.1 c001-interconnect-1   
@@ -27,7 +27,7 @@

 10.10.2.1 login1 login1.tumulus.tld login1-ice1-1
 10.20.2.1 login1-interconnect-1 
- 10.10.102.1 blogin1 
+10.10.102.1 blogin1 


```

Reply:
Will do both.